File: 프로그래머스/3/43162. 네트워크/네트워크.cs
using System;

public class Solution {
    private bool[] visited;
    public int solution(int n, int[,] computers) {
        int answer = 0;
        visited = new bool[n];
        for(int i = 0; i < n; i++)
        {
            if(!visited[i])
            {
                DFS(i, computers);
                answer++;
            }
        }
        return answer;
    }
    
    private void DFS(int n, int[,] computers)
    {
        visited[n] = true;
        for(int i = 0; i < computers.GetLength(0); i++)
        {
            if(i != n && computers[n, i] == 1 && !visited[i])
            {
                DFS(i, computers);
            }
        }
    }
}

File: 프로그래머스/3/43162. 네트워크/네트워크.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

vector<int> visited;
vector<vector<int>> graph;

void dfs(int node, int n)
{
    visited[node] = 1;
    for(int i = 0; i < n; i++)
    {
        if(graph[node][i] == 1 && !visited[i])
        {
            dfs(i, n);
        }
    }
}
int solution(int n, vector<vector<int>> computers) {
    int answer = 0;
    
    graph = computers;
    visited.resize(n, 0);
    
    for(int i = 0; i < n; i++)
    {
        if(!visited[i])
        {
            dfs(i, n);
            answer++;
        }
    }
    return answer;
}